src/tests/zf_unit/netns.sh

```diff
@@ -21,4 +21,36 @@ echo 4096 > /proc/sys/vm/nr_hugepages
 ifconfig lo up
 shift
 
-exec "$@"
+output="$(timeout ${TEST_TIME_OUT:-120} $@)"
+rc="$?"
+echo "$output"
+if [ "$rc" -ne 0 -a -n "$UT_OUTPUT" ]; then
+  if [ ! -f "$UT_OUTPUT" ]; then
+    output_dir="$(dirname $UT_OUTPUT)"
+    if [ -n "$output_dir" -a ! -d "$output_dir" ]; then
+      mkdir -p "$output_dir"
+    fi
+    touch "$UT_OUTPUT"
+  fi
+
+  echo "# Failed \`$(basename $@)\` in \`$TEST_TARGET\`" >> "$UT_OUTPUT"
+  if [ "$rc" -eq 124 ]; then
+    echo "Test timed out (rc=$rc)." >> "$UT_OUTPUT"
+  else
+    echo "Test returned $rc." >> "$UT_OUTPUT"
+  fi
+
+  echo "## Suggested Reproducer" >> "$UT_OUTPUT"
```

Comment:
I wanted to suggest using HereDocs, but a few moments late for a party! (Either way is OK.)

Reply:
From a brief look, that probably would give a much nicer-to-read bit of code here! I created this bit with the intention that it could be updated with a nicer markdown summary, so perhaps something to follow up on when this code is next changed!